### PR TITLE
[TIMOB-24729]:iOS: Resolve iOS 8 deprecations for ABPeoplePickerNavigationController

### PR DESCRIPTION
**Pull request by vijaysingh-axway**
https://jira.appcelerator.org/browse/TIMOB-24729

### CODE REVIEW

File: iphone/Classes/ContactsModule.m [hunk 1]
```diff
@@ -900,131 +900,116 @@ -(void)peoplePickerNavigationControllerDidCancel:(ABPeoplePickerNavigationContro
...
-	}
-	return YES;
-}
 // Called after a person has been selected by the user. New in iOS 8
 - (void)peoplePickerNavigationController:(ABPeoplePickerNavigationController*)peoplePicker didSelectPerson:(ABRecordRef)person
 {
-    [self peoplePickerNavigationController:peoplePicker shouldContinueAfterSelectingPerson:person];
+    if (selectedPersonCallback) {
+        TiContactsPerson* personObject = nil;
+        if ([TiUtils isIOS8OrGreater] && (ABAddressBookGetAuthorizationStatus() != kABAuthorizationStatusAuthorized)) {
```
Comment (hansemannn) [inline iphone/Classes/ContactsModule.m]: `[TiUtils isIOS8OrGreater]` should not be required when a) the delegate only fires for iOS8+ and b) we target iOS 8+.

File: iphone/Classes/ContactsModule.m [hunk 2]
```diff
@@ -900,131 +900,116 @@ -(void)peoplePickerNavigationControllerDidCancel:(ABPeoplePickerNavigationContro
...
+        TiContactsPerson *personObject = [[[TiContactsPerson alloc] _initWithPageContext:[self executionContext] recordId:id_ module:self] autorelease];
+        NSString *propertyName = nil;
+        id value = [NSNull null];
+        id label = [NSNull null];
+        
+        //if statement to handle undocumented ring and text tone property from apple
+        //only implemented in this method, since apple doesn't want people fooling around with these
+        //null values are accompanied. Only inform app that user selected this property in the peoplePicker
+        if (property == appleUndocumentedToneProperty)
+        {
```
Comment (hansemannn) [inline iphone/Classes/ContactsModule.m]: Brackets again :-)

File: iphone/Classes/ContactsModule.m [hunk 3]
```diff
@@ -900,131 +900,116 @@ -(void)peoplePickerNavigationControllerDidCancel:(ABPeoplePickerNavigationContro
...
+                propertyName = @"ringVibration";
+            }
+            if (identifier == appleUndocumentedTextToneIdentifier) {
+                propertyName = @"textTone";
+            }
+            if (identifier == appleUndocumentedTextVibrationIdentifier) {
+                propertyName = @"textVibration";
+            }
+        }
+        else if (identifier == kABMultiValueInvalidIdentifier) {
```
Comment (hansemannn) [inline iphone/Classes/ContactsModule.m]: } else if

**Review by hansemannn — CHANGES_REQUESTED** [review]
Minor changes, thx!

File: iphone/Classes/ContactsModule.m (hunk 1 above)
Comment (vijaysingh-axway) [inline iphone/Classes/ContactsModule.m]: Done.

File: iphone/Classes/ContactsModule.m (hunk 3 above)
Comment (vijaysingh-axway) [inline iphone/Classes/ContactsModule.m]: Done. Thanks

File: iphone/Classes/ContactsModule.m [hunk 4]
```diff
@@ -387,15 +387,13 @@ -(void)showContacts:(id)args
     }
 	picker = [[ABPeoplePickerNavigationController alloc] init];
 	[picker setPeoplePickerDelegate:self];
```
Comment (hansemannn) [inline iphone/Classes/ContactsModule.m]: Can you fix the indentation here and in line 376/377.

**Review by hansemannn — CHANGES_REQUESTED** [review]

File: iphone/Classes/ContactsModule.m (hunk 4 above)
Comment (vijaysingh-axway) [inline iphone/Classes/ContactsModule.m]: In xcode it was shows proper indentation. Maybe some legacy issue. Done.

File: iphone/Classes/ContactsModule.m [hunk 5]
```diff
@@ -373,8 +373,8 @@ -(void)showContacts:(id)args
         if (fields != nil) {
             NSMutableArray* pickerFields = [NSMutableArray arrayWithCapacity:[fields count]];
             for (id field in fields) {
-				id property = nil;
-				if (property = [[[TiContactsPerson iOS9propertyKeys] allKeysForObject:field] objectAtIndex:0]) {
+                id property = nil;
+                if (property = [[[TiContactsPerson iOS9propertyKeys] allKeysForObject:field] objectAtIndex:0]) {
                     [pickerFields addObject:property];
```
Comment (hansemannn) [inline iphone/Classes/ContactsModule.m]: Close to correct :P

**Review by hansemannn — APPROVED** [review]

**Review by htbryant94 — APPROVED** [review]
Approved! ran test case against iOS8/9/10 with Ti SDK 6.1.0.GA & Master.

**Pull request merged**